torch/csrc/distributed/c10d/ProcessGroupGloo.cpp

```diff
@@ -2857,20 +2857,7 @@ void ProcessGroupGloo::monitoredBarrier(
 }
 
 void ProcessGroupGloo::setSequenceNumberForGroup() {
-  if (rank_ == 0) {
-    // Create and broadcast sequence number
-    auto seq = 1 + rand();
-    sequenceNum_ = c10d::SequenceNum(seq);
-    std::vector<char> values = c10d::toVec<char>(seq, kBytes);
-    store_->set(kSeqNumStoreKey, values);
-  } else {
-    // Read rank 0's sequence number from store.
-    sequenceNum_ = c10d::SequenceNum();
-    store_->wait({kSeqNumStoreKey}, options_->timeout);
-    std::vector<char> values = store_->get(kSeqNumStoreKey);
-    uint64_t num = c10d::fromVec<char>(values);
-    sequenceNum_->set(num);
-  }
+  sequenceNum_ = c10d::SequenceNum(0);
```

Comment:
does this still work if there are subgroups? Is sequence number only used for logging?

Reply:
the only issue is a slightly worsened UX for debugging, because if multiple subgroups are there, it might be hard to tell which collective corresponds to which subgroup when there are issues in TDD=DETAIL mode.